Build fails on Linux Mint 'Nadia'

The error:

```
gpi_vpi.c: In function ‘gpi_free_recurring’:
gpi_vpi.c:645:9: error: format ‘%s’ expects a matching ‘char *’ argument [-Werror=format]
```

And my gcc version:

```
gfg@finn-linux ~/source/cocotb $ gcc --version
gcc (Ubuntu/Linaro 4.7.2-2ubuntu1) 4.7.2
Copyright © 2012 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
```

warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

I was able to build fine on Fedora 'Beefy Miracle'. It's probably fixable with some complier flags, but looking at gpi_free_recurring() shows that a printf is used. This should probably be a LOG_CRITICAL.

--- a/lib/vpi_shim/gpi_vpi.c
+++ b/lib/vpi_shim/gpi_vpi.c
@@ -642,7 +642,7 @@ static int gpi_free_recurring(p_vpi_cb_user_data user_data)
     int rc;
     vpiHandle cb_hdl = user_data->cb_hdl;
     if (!cb_hdl) {
-        printf("VPI: %s passed a NULL pointer : ABORTING");
+        LOG_CRITICAL("VPI: passed a NULL pointer : ABORTING");
         exit(1);
     }
 

